exonum/src/api/backends/actix.rs

```diff
@@ -474,28 +502,83 @@ impl From<AllowOrigin> for Cors {
     }
 }
 
-#[test]
-fn allow_origin_from_str() {
-    fn check(text: &str, expected: AllowOrigin) {
-        let from_str = AllowOrigin::from_str(text).unwrap();
-        assert_eq!(from_str, expected);
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn allow_origin_from_str() {
+        fn check(text: &str, expected: AllowOrigin) {
+            let from_str = AllowOrigin::from_str(text).unwrap();
+            assert_eq!(from_str, expected);
+        }
+
+        check(r#"*"#, AllowOrigin::Any);
+        check(
+            r#"http://example.com"#,
+            AllowOrigin::Whitelist(vec!["http://example.com".to_string()]),
+        );
+        check(
+            r#"http://a.org, http://b.org"#,
+            AllowOrigin::Whitelist(vec!["http://a.org".to_string(), "http://b.org".to_string()]),
+        );
+        check(
+            r#"http://a.org, http://b.org, "#,
+            AllowOrigin::Whitelist(vec!["http://a.org".to_string(), "http://b.org".to_string()]),
+        );
+        check(
+            r#"http://a.org,http://b.org"#,
+            AllowOrigin::Whitelist(vec!["http://a.org".to_string(), "http://b.org".to_string()]),
+        );
+    }
+
+    fn assert_responses_eq(left: HttpResponse, right: HttpResponse) {
+        assert_eq!(left.status(), right.status());
+        assert_eq!(left.headers(), right.headers());
+        assert_eq!(left.body(), right.body());
     }
 
-    check(r#"*"#, AllowOrigin::Any);
-    check(
-        r#"http://example.com"#,
-        AllowOrigin::Whitelist(vec!["http://example.com".to_string()]),
-    );
-    check(
-        r#"http://a.org, http://b.org"#,
-        AllowOrigin::Whitelist(vec!["http://a.org".to_string(), "http://b.org".to_string()]),
-    );
-    check(
-        r#"http://a.org, http://b.org, "#,
-        AllowOrigin::Whitelist(vec!["http://a.org".to_string(), "http://b.org".to_string()]),
-    );
-    check(
-        r#"http://a.org,http://b.org"#,
-        AllowOrigin::Whitelist(vec!["http://a.org".to_string(), "http://b.org".to_string()]),
-    );
+    #[test]
+    fn test_create_warning_header() {
+        assert_eq!(
+            &create_warning_header("Description"),
+            "299 - \"Description\""
+        );
+    }
+
+    #[test]
+    fn json_responses() {
+        use chrono::TimeZone;
+
+        let actual_response = json_response(Actuality::Actual, 123);
+        assert_responses_eq(actual_response, HttpResponse::Ok().json(123));
+
+        let deprecated_response_no_deadline = json_response(Actuality::Deprecated(None), 123);
+        let expected_warning_text =
+            "Deprecated API: This endpoint is deprecated, \
+             see the documentation to find an alternative. \
```

Comment:
Now, here's the drawback of the current approach: If a user wants a custom warning text, she would need to create PR in the core / modify core sources / petition the UN / ... Whereas with decorator mechanics, this would be significantly easier, and even adding new functionality to the `Deprecated` decorator (e.g., `description` / `doc_url` method) would likely be non-breaking and easy to migrate to.

Reply:
But should he? Of course, deprecation warnings aren't standardized (yet), but IMO it will be better for end-side users to expect messages from the framework in some certain format, rather than look at the docs for every different service to check if it customizes responses.

Reply:
IMO, "see the documentation" bit sort of asks for a link 🙃 Otherwise, it's unclear which documentation to search for.

Reply:
It's now possible to add an additional description, which will be included to the deprecation warning (thus the beginning of the warning remains well-defined, but additional info can be provided).